File: javascript/leetcode/392. Is Subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let j = 0
    
    for(let i of s) {
        while(j < t.length && t[j] !== i)
            ++j
        
        if(j === t.length) return false
        ++j
    }
    
    return true;
};
 

File: c++/leetcode/392. Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int j = 0;
        for(auto i : s) {
            while(j < t.size() && t[j] != i) 
                ++j;
            
            if(j == t.size()) return false;
            
            ++j;
        }
        
        return true;
    }
};
